serverless/src/lambda/types.ts

```diff
@@ -0,0 +1,22 @@
+import { CFN_IF_FUNCTION_STRING } from "types";
+
+// Array of string ARNs in normal case. Can also include
+// CFN conditional objects represented as e.g.:
+// {"Fn::If": ["isProd", ["prod-layer-arn"], ["stg-layer-arn"]]}
+export type LambdaLayersProperty =
+  | (string | LambdaLayersProperty)[]
+  | { [CFN_IF_FUNCTION_STRING]: [string, LambdaLayersProperty, LambdaLayersProperty] };
+
+export interface FunctionProperties {
```

Comment:
This and some of the other types here look very similar to CDK defined props.  What makes us have our own vs using and extending theirs here?

Reply:
I think it's because unlike CDK, CloudFormation Macro doesn't have a library that exposes these interfaces, resulting in poor typing support.
Actually [the example CloudFormation Macros](https://github.com/aws-cloudformation/aws-cloudformation-templates/tree/main/CloudFormation/MacrosExamples) are all in Python, which has even worse type support. I guess developer experience for CloudFormation Macro is not a priority for AWS teams.

Reply:
I definitely don't completely understand the context of where this gets used, but does us having a type like this that is matching something AWS provides force us to keep parity with them, or are we able to just keep the subset that we need here?

Reply:
I think we are able to just keep the subset that we need here. If we don't need a field, then we don't need to add it to the interface, and there will be no error. It's just harder to access it.
I wish I could talk to Hannah Sun, who built this from scratch (e.g. https://github.com/DataDog/datadog-cloudformation-macro/commit/813d881a0cabc3b25446ea868bbb4f9760d655bb), but she's no longer with Datadog.
Do you have any suggestion on improving the code?

Reply:
I agree then its fine, and probably not in the scope of this PR since you're just moving it.  If it was something that we needed to keep up to date, we could use the CDK lambda function props or the CFN function props by importing it then augmenting whatever we needed to add on top of that

Reply:
I don't think it's something we need to keep up to date because we only need a subset of fields and they need to ensure backward compatibility, but let me add a task to my backlog to explore if it's possible to import CFN function props. I agree it's not pretty to define those interfaces ourselves.